number_theory/prime.h

```diff
@@ -36,10 +36,10 @@ class Sieve {
     is_prime_[1] = false;
     // Use uint64_t to avoid multiplication overflow. Due to the time and space
     // complexity of the algorithm, uint64_t is large enough.
-    for (uint64_t i = 2; i * i <= num_limit_; ++i) {
```

Comment:
It might be more elegant to change `num_limit_`'s type from `T` to `uint64_t` instead of casting it everytime we use. What would you think?

Reply:
I tried to change its type to uint64_t but then it required more casts to T.
So I use a new local variable instead.

Reply:
Looks nice. What about using our `numeric_cast` instead of `static_cast`?